matrix/src/stack.rs

```diff
@@ -3,111 +3,143 @@ use core::ops::Deref;
 
 use crate::Matrix;
 
-/// A combination of two matrices, stacked together vertically.
+/// A matrix composed by stacking two matrices vertically, one on top of the other.
+///
+/// Both matrices must have the same `width`.
+/// The resulting matrix has dimensions:
+/// - `width`: The same as the inputs.
+/// - `height`: The sum of the `heights` of the input matrices.
+///
+/// Element access and iteration will first access the rows of the top (`first`) matrix,
+/// followed by the rows of the bottom (`second`) matrix.
```

Comment:
Can clean this up now after the renaming
```suggestion
/// Element access and iteration will first access the rows of the top matrix,
/// followed by the rows of the bottom matrix.
```